HHS/Head-Start-TTADP
# [TTAHUB-2128] [TTAHUB-2220] Goal creator FE&BE

## Description of change



## How to test

- Create a TR goal (event with goal text, then create a session for that event).
- Validate the creator is YOU (GoalCollaborators table).
- Edit the event, add some POCs to the list. Ensure the goal creator in the same table has now been updated to the first POC in the list (sorted alphabetically). Remove the POC and it will update back to you.

- There's already logic in place to populate creator/linker in hooks, so validate that other goals from a recent dump have their "Entered by" populated in goal cards, and if the user has two roles, both are listed:


![Screenshot 2024-02-27 at 10 02 11 AM](https://github.com/HHS/Head-Start-TTADP/assets/17074357/2754acc3-084a-4304-bd29-8ae6937dff27)

Hover the role to see the user's name.

- Make sure the "entered by role" filter works as expected:

![Screenshot 2024-02-27 at 10 03 10 AM](https://github.com/HHS/Head-Start-TTADP/assets/17074357/27672c04-bed7-428f-866a-7b3f230206f2)

In the case of the goal rollup, where two goals share a card, ensure two creators are shown:

![Screenshot 2024-02-27 at 1 58 46 PM](https://github.com/HHS/Head-Start-TTADP/assets/17074357/fdee5cb9-0d03-4e88-98e9-59ca55b542c4)


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2128


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
